File: 2021/js/day0101.js
/**
 * Author: Loading BG#7962
 */
'use strict';

const nthEven = n => n * 2 - 2;

const validateEmail = email => /.+@.+\..+/.test(email);

function atbashLetter(l) {
    if (l.toLowerCase() === l.toUpperCase()) {
        return l;
    } else if (l === l.toLowerCase()) {
        return String.fromCharCode('z'.charCodeAt(0) - l.charCodeAt(0) + 'a'.charCodeAt(0));
    } else {
        return String.fromCharCode('Z'.charCodeAt(0) - l.charCodeAt(0) + 'A'.charCodeAt(0));
    }
}

const atbash = sentence => sentence.split("").map(atbashLetter).join("");

console.log(`nth_even #1: ${nthEven(1)}`);
console.log(`nth_even #2: ${nthEven(2)}`);
console.log(`nth_even #3: ${nthEven(100)}`);
console.log("-----------");
console.log(`validateEmail #1: ${validateEmail("@gmail.com")}`);
console.log(`validateEmail #2: ${validateEmail("hello.gmail@com")}`);
console.log(`validateEmail #3: ${validateEmail("gmail")}`);
console.log(`validateEmail #4: ${validateEmail("hello@gmail")}`);
console.log(`validateEmail #5: ${validateEmail("hello@edabit.com")}`);
console.log("-----------");
console.log(`atbash #1: ${atbash("apple")}`);
console.log(`atbash #2: ${atbash("Hello world!")}`);
console.log(`atbash #3: ${atbash("Christmas is the 25th of December")}`);

File: 2021/elixir/Day0101.ex
#
# Author: Loading BG#7962
#

defmodule Day0101 do
  def nth_even(n), do: 2 * n - 2
  
  def validate_email(email), do: String.match? email, ~r/^.+@.+\..+$/
  
  defp atbash_letter(l) when l in ?a..?z, do: ?z - l + ?a
  defp atbash_letter(l) when l in ?A..?Z, do: ?Z - l + ?A
  defp atbash_letter(l), do: l
  def atbash(s), do: String.to_charlist(s)
                       |> Enum.map(&atbash_letter &1)
                       |> List.to_string
end

IO.inspect Day0101.nth_even(1), label: "nth_even #1"
IO.inspect Day0101.nth_even(2), label: "nth_even #2"
IO.inspect Day0101.nth_even(100), label: "nth_even #3"
IO.puts "---------"
IO.inspect Day0101.validate_email("@gmail.com"), label: "validateEmail #1"
IO.inspect Day0101.validate_email("hello.gmail@com"), label: "validateEmail #2"
IO.inspect Day0101.validate_email("gmail"), label: "validateEmail #3"
IO.inspect Day0101.validate_email("hello@gmail"), label: "validateEmail #4"
IO.inspect Day0101.validate_email("hello@edabit.com"), label: "validateEmail #5"
IO.puts "---------"
IO.inspect Day0101.atbash("apple"), label: "atbash #1"
IO.inspect Day0101.atbash("Hello world!"), label: "atbash #2"
IO.inspect Day0101.atbash("Christmas is the 25th of December"), label: "atbash #3"
